File: MyungJi/231127_week/programmers_161989_덧칠하기.cpp
#include <bits/stdc++.h>

using namespace std;
bool colored[100004];

int solution(int n, int m, vector<int> section) {
    int answer = 0;
    //1. section 시작점부터 m칸 색칠
    for(int i=0;i<section.size();i++){
        int start = section[i];
        if (colored[start]==1) continue;
        for(int j=start;j<min(start+m, n+1);j++){
            colored[j]=true;  
        } 
        answer++;
    }
    return answer;
}

File: MyungJi/240304_week/programmers_161989_덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 1;
        int pos = section[0];
        for (int i=1;i<section.length;i++){
            if (pos+m-1 < section[i]){
                answer++;
                pos = section[i];
            }
        }

        return answer;
    }
}